src/Simplex/Chat.hs

```diff
@@ -231,48 +231,120 @@ processAgentMessage user@User {userId, profile} agentConnId agentMessage = do
-              forM_ (filter (\m' -> memberStatus m' >= GSMemConnected) . map fst $ members) $ \m' ->
-                sendDirectMessage agentConnId $ XGrpMemIntro (memberId m') (memberRole m') (memberProfile m')
-            _ -> do
-              if Just (invitedBy membership) == (IBContact <$> memberContactId m)
-                then do
-                  -- sender invited the current user
-                  withStore $ \st -> updateGroupMemberStatus st userId (groupMemberId m) GSMemConnected
-                  showUserConnectedToGroup gName
-                  pure ()
-                else do
-                  showConnectedGroupMember gName $ displayName (memberProfile m :: Profile)
+              mIds <- withStore $ \st -> createIntroductions st m
+              sendGroupMessage members . (`XGrpMemNew` mIds) $ memberInfo m
+              forM_ (filter memberActive . map fst $ members) $ \m' ->
+                sendDirectMessage agentConnId . XGrpMemIntro $ memberInfo m'
+            GCPreMember -> do
+              -- TODO send probe and decide whether to use existing contact connection or the new contact connection
+              -- TODO notify member who forwarded introduction - question - where it is stored? There is via_contact but probably there should be via_member in group_members table
+              showConnectedGroupMember gName $ displayName (memberProfile m :: Profile)
+              pure ()
+            GCPostMember -> do
+              -- TODO notify member who forwarded introduction - question - where it is stored? There is via_contact but probably there should be via_member in group_members table
+              showConnectedGroupMember gName $ displayName (memberProfile m :: Profile)
+              pure ()
+            GCHostMember -> do
+              showUserConnectedToGroup gName
+              pure ()
+            GCUserMember -> do
+              -- TODO implementation error - log
+              pure ()
         MSG meta msgBody -> do
           ChatMessage {chatMsgEvent} <- liftEither $ parseChatMessage msgBody
           case chatMsgEvent of
-            XGrpMemNew memId memRole memProfile -> do
-              Group {membership, members} <- withStore $ \st -> getGroup st user gName
-              when (memberId membership /= memId && isNothing (find ((== memId) . memberId . fst) members)) $
-                withStore $ \st -> pure () -- add new member as GSMemAccepted
+            XGrpMemNew memInfo@(MemberInfo memId _ _) _mIds -> do
+              group@Group {membership} <- withStore $ \st -> getGroup st user gName
+              case memberCategory m of
+                GCHostMember -> do
+                  -- TODO add listed members to some list of expected members - group_members table is not very good as it will make many fields nullable
+                  -- TODO maybe we should not set user to GSMemAnnounced to avoid inconsistency when for other member it means not connected yet...
```

Comment:
it wouldn't mean user is connected to other members for the user himself as well.. I think it's okay to assign this status here